sessionctx/variable/sysvar.go

```diff
@@ -139,7 +139,7 @@ var defaultSysVars = []*SysVar{
 	{Scope: ScopeGlobal | ScopeSession, Name: CharacterSetClient, Value: mysql.DefaultCharset, skipInit: true, Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
 		return checkCharacterSet(normalizedValue, CharacterSetClient)
 	}},
-	{Scope: ScopeNone, Name: Port, Value: "4000", Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxUint16},
+	{Scope: ScopeGlobal, Name: Port, Value: "4000", Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxUint16},
```

Comment:
Sorry, this is more complex :( Otherwise I would have fixed the bug when I reported it.

To start with it needs to be read-only still:

```suggestion
	{Scope: ScopeGlobal, Name: Port, Value: "4000", Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxUint16, ReadOnly: true},
```

But the bigger problem is that it is set when the server starts up:
https://github.com/pingcap/tidb/blob/db3e69eb68de9acfcebd69909e7309caed64914c/tidb-server/main.go#L585

We need to make sure that Set only occurs to the local TiDB instance, since each tidb-server in the cluster could be listening on a different port. However, currently the set will propagate to other tidb servers.

There is a proposal which will fix this shortly: https://github.com/pingcap/tidb/blob/master/docs/design/2021-12-08-instance-scope.md

Reply:
When do we plan to support the feature of "instance scope"? Shall we fix the current issue first, and then deal with it when the feature of "Instance Scope" is supported?
Step 1 in the design document is quite convenient to achieve, but there are some compatibility problems that need to be explained.

Reply:
@CbcWestwolf is working on instance scope now. The compatibility issues in the proposal have already been addressed, as we have switched all settings which use Instance + Global to now only use one or the other.

Reply:
We have a PR available for it now: https://github.com/pingcap/tidb/pull/32888